kornia/core/tensor_wrapper.py

```diff
@@ -0,0 +1,142 @@
+# pytorch tensor wrapper class
+# insipired by:
+# https://github.com/pytorch/pytorch/blob/591dfffa38848de54b7f5f4e49260847024c9281/test/test_overrides.py#L748
+import collections
+from typing import Any
+
+import torch
+from torch import Tensor
+
+# wrap inputs if necessary
+
+# TODO: promote to KORNIA_WRAP
+
+
+def wrap(v, cls):
+    if type(v) in {tuple, list}:
+        return type(v)(wrap(vi, cls) for vi in v)
+
+    return cls(v) if isinstance(v, Tensor) else v
+
+
+# TODO: promote to KORNIA_UNWRAP
+
+
+def unwrap(v):
+    if type(v) in {tuple, list}:
+        return type(v)(unwrap(vi) for vi in v)
+
+    return v._data if not isinstance(v, Tensor) else v
+
+
+class TensorWrapper:
```

Comment:
Maybe, in the future, we can use `TensorWrapper` to represent things like `BBox`?

Reply:
i think yes, but we have to be careful because TensorWrapper exposes everything from a tensor and in some cases maybe it doesn't make sense or lead to bugs. For things like Vector, Image, Matrix, for sure; BoundingBox not very clear if it's easy to just overwrite what is needed (or maybe filter via Tensorwrapper)

Reply:
the cool thing is that `TensorWrapper` can have methods like `__array__` to directly interface to numpy

Reply:
and ideally should be able to ingest numpy arrays or others as datastorage

Reply:
Ingest numpy array or others, but cast/transform to be a tensor?

Reply:
casto to the corresponding main backend array structure: torch->Tensor, numpy->ndarray, etc

Reply:
gotcha it